airflow/datasets/__init__.py

```diff
@@ -159,6 +161,20 @@ def iter_datasets(self) -> Iterator[tuple[str, Dataset]]:
         raise NotImplementedError
 
 
+@attr.define()
+class DatasetAlias:
```

Comment:
I wonder if it’s a good idea to make this a subclass of BaseDataset. Would this make typing a bit easier, and reduce some conditionals? We could add some new abstract methods if needed.

Reply:
Update it to a subclass of BaseDataset. Not as useful for now, but not harm either. might be useful when I create the next PR to add scheduling support